File: tests/fixtures/kt/sample.kt
package com.myapp.core

import kotlinx.coroutines.Deferred

/**
 * Person data class.
 */
data class Person(val name: String, val age: Int) {
    val greeting: String get() = "Hello, $name"

    suspend fun processAsync(id: Int, data: String): Result {
        return doWork(id, data)
    }

    override fun toString(): String = "$name ($age)"
}

/**
 * Generic repository interface.
 */
interface Repository<T> {
    suspend fun getById(id: Int): T
    suspend fun save(entity: T)
}

enum class Status { ACTIVE, INACTIVE, PENDING }

/**
 * Sealed class example.
 */
sealed class NetworkResult<out T> {
    data class Success<T>(val data: T) : NetworkResult<T>()
    data class Error(val message: String) : NetworkResult<Nothing>()
}

object PersonFactory {
    fun create(name: String, age: Int) = Person(name, age)
}

fun String.greet(): String = "Hi, $this!"

File: tests/fixtures/swift/sample.swift
import Foundation

/// Person class.
class Person {
    var name: String
    var age: Int

    init(name: String, age: Int) {
        self.name = name
        self.age = age
    }

    func greet() -> String {
        return "Hello, \(name)"
    }

    func processAsync(id: Int, data: String) async -> Result {
        return await doWork(id: id, data: data)
    }
}

/// Repository protocol.
protocol Repository {
    associatedtype T
    func getById(id: Int) async throws -> T
    func save(entity: T) async throws
}

/// Status enum.
enum Status {
    case active
    case inactive
    case pending
}

/// Person struct.
struct PersonDto {
    let name: String
    let age: Int
}

/// PersonFactory extension on Person.
extension Person {
    static func create(name: String, age: Int) -> Person {
        return Person(name: name, age: age)
    }
}
